app/(student)/point-allocations/page.jsx

```diff
@@ -1,7 +1,311 @@
-import React from "react";
+"use client"
+import React, { useState } from 'react'
+import { BsThreeDotsVertical } from 'react-icons/bs';
+import {VscSettings} from 'react-icons/vsc'
+import {BiSolidReport} from 'react-icons/bi'
+import {AiFillEye, AiOutlineCaretDown, AiOutlineCaretUp} from 'react-icons/ai'
+import {BsXLg} from 'react-icons/bs'
+import {MdArrowDownward, MdArrowUpward} from 'react-icons/md'
 
-const PointAllocations = () => {
-  return <div>PointAllocations</div>;
-};
+ const PointAllocation = () => {
+  const initialIsOpenState = Array(data.length).fill(false);
+  const [isOpen, setIsOpen] = useState(initialIsOpenState);
+  const [selectedTransaction, setSelectedTransaction] = useState(null);
+  const [isToggled, setIsToggled] = useState(false);
+  const [isClicked, setIsClicked] = useState(false);
+  const [isUp, setIsUp] = useState(false);
+  const [isSort, setIsSort] = useState(false);
 
-export default PointAllocations;
+  const toggleTransaction = (index) => {
+    setIsOpen((prev) => {
+      const newState = [...prev];
+      newState[index] = !newState[index];
+      return newState;
+    });
```

Comment:
pwede to pasimplehin like
```suggestion
  const toggleTransaction = (index) => {
    setIsOpen((prev) => !prev);
```

Reply:
pre sinunod ko yan kaso biglang hindi nag function yung threedots sa table

Reply:
ano error